### ISSUE TITLE
UI overhaul ideas wanted?

### ISSUE THREAD

**Issue by natrius** [renyuneyun/Easer#187]
I wanted to ask, do you want suggestions for a better UI, a changed UI or soemthing like that? Some could be a little bit, some could be massive.

Do you want to get something like that or better not?

**Comment by renyuneyun** [comment]
Absolutely!
I totally agree that the UI is not good (as said in the app description on f-droid), and do plan to improve it (also see [TODO](https://github.com/renyuneyun/Easer/blob/master/docs/en/TODO.md)).

What I have considered mainly focus on two parts:
1. Better representation of the Script tree (and Script content)
  * A temporary method would be #181 
  * Eventually a graphical view should be presented
2. Better UI for editing Script/Event/Condition/Profile

But suggestions for all aspects are welcome.
Though the main focus currently is the functionality, UI improvement is also a thing.

**Comment by churib** [comment]
It does not really belong to this topic, but it would be great to have DSL in which you can describe the tree.
The DSL scripts can be easily edited on a desktop computer, copied to the phone and imported, visualized and executed by Easer.

Btw: many thanks for this greatly evolving app and your affords!

**Comment by aasami** [comment]
Have you ever heard of Llama? Llama - location profiles.
I think their UI is great! Clean, functional, intuitive. Why to reinvent the wheel, when we can inspire us from existing apps? Try it out, and let me know what you think.

**Comment by natrius** [comment]
Uh, thanks for bringing this up again :D

What i thought is: 
When i open the app, i want to see what Actions i have set. The title i gave, maybe with some smaller written description below to get a fast overview. Maybe even marked, which of the Actions/Events/Scripts is active right now (mark the action/event/script green for example). And a switch on the right side to activate/deactivate them fast and without hassle. Grey out the whole Event/script/Action when inactive.

To add a new Action, press a + on the bottom right corner (similar to new message in any messenger, etc.) and now follows a 3 step-creation-process. 

 - First screen [Description]: Add Action name and optional a description (example: Car mode // Bluetooth, GPS, Brightness)
 - Second screen [Actions]: what needs to be triggered (i keep using that word :D ), so Bluetooth, GPS and Screen brightness. Basically everything thats now in "profile"
- Third screen [Triggers and Conditions]: when it should be triggered. Time, NFC tag, Location, battery level. Now "events" and "conditions" combined. Or split in two steps. 

Right now i will set up one profile for one script, example for carmode. I will put in there some different things to toggle and i will set up another profile for another script because most probably i can't use the previous profile anyway. So the menu will get stuffed. Or i split the profile up to the single things to trigger, but then its stuffed anyway. 
The way i mentioned it above there is one menu that will get filled with exactly i want - my actions. And i can read fast if its active now, deactivated at all and if i have written a short description, thats there too. 
Everything else is exactly where it should be. 

I completely understand if you do not like that approach. :)

**Comment by racuna** [comment]
Hi, I liked pivot to "sort of" debug my scripts. Are you planning to add more stuff to it? Like different colors if the box is "active" (condition satisfied) or some kind of interaction: "tap to open"

thanks,
racuna

**Comment by renyuneyun** [comment]
@racuna Yes of course.

Actually the idea of that pivot page is to gradually replace other pages (Script, Event, Condition, Profile), at lease making them "secondary" or "advanced".

**Comment by racuna** [comment]
Awesome. 

So, events are still a thing?

**Comment by renyuneyun** [comment]
@racuna Yes, why wouldn't? It will always be a thing as a mechanism, until some (if ever) other mechanism succeeds both Event and Condition.

**Comment by PopeRigby** [comment]
I think a dark mode is also important for the UX

**Comment by ildar** [comment]
Hi, I'm back after a long break ))
Seen "pivot" for the first time. Nice picture! Yet why its name "pivot"?
Maybe "graph" or "diagram" would be simpler and easier?
